message/tests.py

```diff
@@ -81,3 +84,93 @@ def test_create_message_with_empty_text(self, player, game_event):
+        client.force_login(player.user)
+        game_event_id = game_event.id
+        url = reverse('process_answer_game_event', args=[game_event_id])
+        client.post(url, {'answer': 'yes'})
+        count_before = Message.objects.count()
+        url = reverse('save_text_message', args=[game_event_id])
+        response = client.post(url, {'mytextbox': 'Test message'})
+
+        assert response.status_code == 302
+        assert Message.objects.count() == count_before + 1
+        message = Message.objects.last()
+        assert message.user_id == player
+        assert message.game_event_id == game_event
+        assert message.text == 'Test message'
+
+    def test_messages_persist_after_join_and_remove(self, client, game_event, player):
+        client.force_login(player.user)
+        game_event_id = game_event.id
+        url = reverse('process_answer_game_event', args=[game_event_id])
+        response = client.post(url, {'answer': 'yes'})
+        assert response.status_code == 200
+        url = reverse('save_text_message', args=[game_event_id])
+        client.post(url, {'mytextbox': 'hi from user'})
+        reverse('remove_from_event', args=[game_event_id])
+        player_messages = [entry.text for entry in Message.objects.filter(game_event_id=game_event)]
+        assert 'hi from user' in player_messages
+
+    def test_success_create_message_function(self, game_event, player):
+        text = 'test message'
+        GameEventPlayer.objects.create(game_event=game_event, player=player)
+        count_before = Message.objects.count()
+        message = Message.create(player, game_event, text)
+        assert Message.objects.count() == count_before + 1
+        assert message.text == 'test message'
+
+    def test_fail_create_message_function(self, game_event, player):
+        text = 'test message'
+        count_before = Message.objects.count()
+        with pytest.raises(ValidationError):
+            Message.create(player, game_event, text)
+        assert Message.objects.count() == count_before
+
+    def test_success_validate_message(self, game_event, player):
+        GameEventPlayer.objects.create(game_event=game_event, player=player)
+        text = 'test message'
+        e = Message.validate_message(player, game_event, text)
+        assert e is None
+
+    @pytest.mark.parametrize(
+        "text, expected_errors",
+        [
+            ('test message', ["Just players from this event can send messages on chat"]),
+            ('', ["Message can't be empty", "Just players from this event can send messages on chat"]),
+            ('           ', ["Message can't be empty", "Just players from this event can send messages on chat"]),
+        ]
+    )
+    def test_fail_validate_message(self, game_event, player, text, expected_errors):
+        with pytest.raises(ValidationError) as current_errors:
+            Message.validate_message(player, game_event, text)
+        errors = "\n".join(current_errors.value)
```

Comment:
What do we have in `current_errors.value`? Looking at what we're putting there, I would not expect there would be anything for which `join` is applicable.

Reply:
> expect

the current_errors.value is this: ["Just players from this event can send messages on chat\nMessage can't be empty"]
on the validation function, I convert the list to single string, and the errors separated by '\n' (I used the join function there)
If on this test I don't use the join it did me this assertion error: 
`E   assert "Just players from this event can send messages on chat\nMessage can't be empty" in ["Message can't be empty", 'Just players from this event can send messages on chat']

message/tests.py:155: AssertionError`

Reply:
> What do we have in `current_errors.value`? Looking at what we're putting there, I would not expect there would be anything for which `join` is applicable.

the current_errors.value is this: ["Just players from this event can send messages on chat\nMessage can't be empty"]
on the validation function, I convert the list to single string, and the errors separated by '\n' (I used the join function there)
If on this test I don't use the join it did me this assertion error:
`E assert "Just players from this event can send messages on chat\nMessage can't be empty" in ["Message can't be empty", 'Just players from this event can send messages on chat']

message/tests.py:155: AssertionError`

Reply:
The outcome of running join on that is just getting the first element of that list - which is the only element.
```python
In [2]: a = ["Just players from this event can send messages on chat\nMessage can't be empty"]

In [3]: "\n".join(a)
Out[3]: "Just players from this event can send messages on chat\nMessage can't be empty"

In [4]: "\n".join(a) == a[0]
Out[4]: True
```
In other words, you can use `[current_errors.value]` instead of what you put into `errors`